### ISSUE TITLE
Visual Studio build broken

### ISSUE THREAD

**Issue by ZwergNaseXXL** [miloyip/nativejson-benchmark#126]
Just tried to get this to build for two hours, still no luck. Used various VS tooling configs between 2015 and 2019. Always getting hundreds of warnings and errors, don't even know where to start. It's probably some mysterious #define somewhere.

Docu is also lacking. Says "just start the thing generated by premake5", but there are two solutions, jsonstat and benchmark. Which is what? Which one should I care about?

I'd be happy to help out with fixes here, but I need some hints.

**Comment by mloskot** [comment]
Likely, this is duplicate of #102 and #123

IOW, if those issues are not fixed, this issue will not be fixed either.

**Comment by ZwergNaseXXL** [comment]
Yeah, I was aware of those issues but didn't get to try it at that time. Now I finally got around to try once more and succeeded, in a limited way. 

My intention was to pitch our in-house JSON lib against the three other ones I know, namely Jansson, nlohmann and RapidJSON. Limiting myself to these three libs made it fairly easy:

- delete everything from src/cjsonlibs/, src/tests/ and thirdparty/ that isn't related to those 3 libs
- comment out those two ULib lines in build/premake5.lua, as suggested in #102 
- comment out the #ifdef in the first line of src/tests/nlohmanntest.cpp, which excludes it from being built by MSVC
- run premake.bat in build/
- open build/vs2015/jsonstat.sln with VS2019, click yes to upgrade everything, build with F7
- same for build/vs2015/benchmark.sln
- set "nativejson" as the startup project, hit Ctrl+F5 to run the tests
- find the test results in result/

Now we just need someone to eliminate the need for make & php in order to transform those results to HTML... :) Until then, looking at the .md files in your favorite editor (there are rendering plugins for Notepad++ and VS) and importing & formatting the .csv files in Excel will do.
